### DISCUSSION TITLE
Postgres Listen/Notify

### DISCUSSION THREAD

**Discussion by 3mIn4** [MaterializeInc/materialize#11254] [Q&A] [answered]
I am new to Materialize, but I am curious: Is it possible to use Postgres Listen/Notify logic and create triggers upon materialized views.
E.g: 

CREATE MATERIALIZED VIEW avg_bid AS
SELECT symbol,
       AVG(bid_price) AS avg
FROM market_orders
GROUP BY symbol;

CREATE OR REPLACE FUNCTION notify_channel_avg_bid() RETURNS TRIGGER AS 
    $$ DECLARE payload JSON;
    BEGIN
    IF (TG_OP = 'DELETE') THEN
        payload = json_build_object('action', 'DELETE', 'symbol', OLD.id);
    ELSIF (TG_OP = 'UPDATE') THEN
        payload = json_build_object('action', 'UPDATE', 'symbol', NEW.id);
    ELSIF (TG_OP = 'INSERT') THEN
        payload = json_build_object('action', 'INSERT', 'symbol', NEW.id);
    END IF;
    PERFORM pg_notify('avg_bid_channel', payload::text);
    RETURN NULL;
    END;
    $$ LANGUAGE plpgsql;
    
  CREATE TRIGGER avg_bid_notify_trigger AFTER INSERT OR UPDATE OR DELETE ON avg_bid
    FOR EACH ROW EXECUTE PROCEDURE notify_channel_avg_bid();

**Comment by joacoc** [comment, marked as answer]
Hey @3mIn4!
Have you checked out [tails](https://materialize.com/docs/sql/tail/#conceptual-framework)? They provide a similar concept to the Postgres Listen/Notify logic. Super suitable for these cases.
